common/exception/ErrorMessage.java

```diff
@@ -556,6 +566,30 @@ public static class TypeWrite extends ErrorMessage {
         }
     }
 
+    public static class Expression extends ErrorMessage {
+        public static final Expression OPERATION_NOT_FOUND =
+                new Expression(1, "The operation could not be resolved for the data-types: '%s' %s '%s'.");
+        public static final Expression FUNCTION_NOT_FOUND =
+                new Expression(2, "The function could not be resolved for the signature: '%s(%s)'.");
+        public static final Expression AMBIGUOUS_ARGUMENT_TYPE =
+                new Expression(3, "The variable '%s' can have multiple value types: [%s].");
+        public static final Expression EVALUATION_ERROR =
+                new Expression(4, "An error occured while evaluating an expression:\nExpression: '%s = %s'; \nInput: '%s'.\nError: '%s'");
+        public static final Expression INVALID_RESULT =
+                new Expression(5, "An operation produced an invalid result: '%s'");
+        public static final Expression INVALID_VALUE_CASTING =
+                new Expression(6, "Invalid value conversion from '%s' to '%s'.");
+        public static final Expression INVALID_VALUE_IID_CASTING =
+                new Expression(7, "Invalid Thing IID casting to '%s'.");
```

Comment:
The key point is that these errors to expressions at all - this should be in the something analogous to the `ThingRead` error message section.

I think you need a new section around the `Thing` errors called `Value`, which includes these two errors inside.

Reply:
eg. `VALUE_CASTING` and `VALUE_IID_CASTING`

Reply:
@krishnangovindraj  i think this still needs implementing